### PR TITLE
GH-264 - Do Not Allow Report Authors to Create Reports When Their Assigned Advisor Position Has an Inactive Status

### PR DESCRIPTION
**Pull request by evdhoudt**
Closes NCI-Agency/anet#264

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -86,6 +86,7 @@
     "lodash.debounce": "4.0.8",
     "lodash.get": "4.4.2",
     "lodash.includes": "4.3.0",
+    "lodash.isempty": "^4.4.0",
```
Comment (VassilIordanov) [inline client/package.json]: Use exact package versions

File: client/src/pages/reports/Form.js [hunk 2]
```diff
@@ -133,16 +145,26 @@ export default class ReportForm extends ValidatableFormWrapper {
...
 
 		const {ValidatableForm, RequiredField} = this
+		const submitText = currentUser.hasActivePosition() ? 'Preview and submit' : 'Save draft'
+		const alertStyle = {top:132, marginBottom: '1rem', textAlign: 'center'}
 
-		const submitText = currentUser.hasAssignedPosition() ? 'Preview and submit' : 'Save draft'
-
+		const supportEmail = dict.lookup('SUPPORT_EMAIL_ADDR')
+		const supportEmailMessage = supportEmail ? `(${supportEmail})` : ''
+		const warningMessageNoPosition = `You do not have an active position and cannot submit a report, please contact the support team ${supportEmailMessage} and request them to set your position status to active`
```
Comment (VassilIordanov) [inline client/src/pages/reports/Form.js]: I would change this to `You do not have an active position and cannot submit a report, please contact your organization's super users and request them to set your position status to active. If you are unsure, you can also contact the support team ${supportEmailMessage}`
@dspCarlson can you take a look?

File: client/src/pages/reports/Show.js [hunk 3]
```diff
@@ -97,7 +97,9 @@ export default class ReportShow extends Page {
 	}
 
 	renderNoPositionAssignedText() {
-		return <p>Notice: This report cannot be submitted because you do not have an assigned position.<br /> -- Please contact your administrator --</p>
+		const supportEmail = dict.lookup('SUPPORT_EMAIL_ADDR')
+		const supportEmailMessage = supportEmail ? `at ${supportEmail}` : ''
+		return <div className="alert alert-warning">Notice: This report cannot be submitted because you do not have an assigned or active position.<br /> -- please contact the support team {supportEmailMessage} --</div>
```
Comment (VassilIordanov) [inline client/src/pages/reports/Show.js]: I suggest something like `... please contact your organization's super users and request them to assign you to a position. If you are unsure, you can also contact  the support team ${supportEmailMessage}` @dspCarlson

File: client/src/pages/reports/Form.js (hunk 2 above)
Comment (dspCarlson) [reply client/src/pages/reports/Form.js]: My suggestion: You cannot submit a report.  Your assigned advisor position has an inactive status.  Please contact your organization's super users {some magic to identify the superusers} and request that they set your position status to active.

File: client/src/pages/reports/Show.js (hunk 3 above)
Comment (dspCarlson) [reply client/src/pages/reports/Show.js]: My suggestion: You cannot submit a report.  You are not assigned to an advisor position.  Please contact your organization's super users {some magic to identify the superusers} and request that they assign you to an advisor position.

Comment (VassilIordanov) [reply client/src/pages/reports/Show.js]: the issue here, at least for now, is that we can't identify the superusers until the user is assigned to a position

**Pull request merged**